### DISCUSSION TITLE
Line art width

### DISCUSSION THREAD

**Discussion by josemoralescc** [bnpr/Malt#222] [Q&A] [answered]
![Screenshot from 2022-01-19 00-29-42](https://user-images.githubusercontent.com/21679398/150035302-56792d5b-929e-42bd-b927-48ea8fe244a4.png)
Hi everyone,

How can I create a line art with a specific width? Like two millimeters for example.

Maybe it's easy but I've been trying for hours. I can only change line width a bit by trial and error.

I have problems with these points:
1. Viewport line width seems to have no relation to rendering.
2. Viewport line width changes with the zoom (doesn't change proportionally with the object).
3. It seems that there is an implicit limit for the value of line width.

Thanks a lot!

**Comment by pragma37** [comment, marked as answer]
Hi!

Line width is specified in pixel units. So if you render at different resolutions the relative size will change.

If you want to set the line width in world units you can divide the intended width by the size of a pixel in world space:
![imagen](https://user-images.githubusercontent.com/36610999/150152448-e53dfaf7-f8dd-43de-9a8d-3e9e7a88f416.png)

If you just want to scale the line width based on render resolution then:
![imagen](https://user-images.githubusercontent.com/36610999/150153333-aa5954d0-23fb-4277-8595-4ec9728a002b.png)

The max line width limit can be changed in the World panel (Malt Settings > Line > Max Width).
(Or in the Line Render node in the upcoming 1.0 version)

**Reply by josemoralescc** [reply]
![Screenshot from 2022-01-20 14-05-17](https://user-images.githubusercontent.com/21679398/150344346-90572f4a-d6f6-409f-b129-0c5affb4e07e.png)
Hi Miguel, thank you for your help!

I've been testing and everything works better.

Is it possible to combine both approaches? Could a line art work (visually) the same way as a Blender object?

**Reply by pragma37** [reply]
> Could a line art work (visually) the same way as a Blender object?

I'm not sure about what do you mean. Doesn't the first approach already work like that?

**Reply by josemoralescc** [reply]
![Screenshot from 2022-01-20 16-45-37](https://user-images.githubusercontent.com/21679398/150372358-199f05cf-1d4d-4fe3-95e7-789a2e8a28d1.png)

Yes, it does. I've had a problem because I didn't adjust Max width in the right way, so sometimes viewport and rendering were not the same.

Now it works, thanks!

**Reply by krigeta** [reply]
> Hi!
> 
> Line width is specified in pixel units. So if you render at different resolutions the relative size will change.
> 
> If you want to set the line width in world units you can divide the intended width by the size of a pixel in world space: ![imagen](https://user-images.githubusercontent.com/36610999/150152448-e53dfaf7-f8dd-43de-9a8d-3e9e7a88f416.png)
> 
> If you just want to scale the line width based on render resolution then: ![imagen](https://user-images.githubusercontent.com/36610999/150153333-aa5954d0-23fb-4277-8595-4ec9728a002b.png)
> 
> The max line width limit can be changed in the World panel (Malt Settings > Line > Max Width). (Or in the Line Render node in the upcoming 1.0 version)

Hello, I am using this method but somehow the viewport and Render image outline are not matching and I have to change the viewport resolution too, isn't there a technique so we can able to make the render and viewport outline(same distance as in Render) look same?
